[Bug]: Direct load API occasionally has an error: direct load in ob version 0.0.0.0 is not supported, minimum version required is 4.2.1.7

### Describe the bug

Direct load API occasionally throws an exception: direct load in ob version 0.0.0.0 is not supported, minimum version required is 4.2.1.7

Executed multiple times, most of the time it can be executed successfully, occasionally the above exception is thrown.

```
Caused by: java.lang.RuntimeException: Fail to obtain direct-load connection.
	at com.oceanbase.spark.directload.DirectLoaderBuilder.build(DirectLoaderBuilder.java:180)
	at com.oceanbase.spark.directload.DirectLoadUtils.buildDirectLoaderFromSetting(DirectLoadUtils.java:46)
	... 13 more
Caused by: com.alipay.oceanbase.rpc.direct_load.exception.ObDirectLoadNotSupportedException: direct load in ob version 0.0.0.0 is not supported, minimum version required is 4.2.1.7
	at com.alipay.oceanbase.rpc.direct_load.protocol.ObDirectLoadProtocolFactory.getProtocol(ObDirectLoadProtocolFactory.java:81)
	at com.alipay.oceanbase.rpc.direct_load.ObDirectLoadConnection.initProtocol(ObDirectLoadConnection.java:188)
	at com.alipay.oceanbase.rpc.direct_load.ObDirectLoadConnection.init(ObDirectLoadConnection.java:87)
	at com.alipay.oceanbase.rpc.direct_load.ObDirectLoadConnectionFactory.buildConnection(ObDirectLoadConnectionFactory.java:44)
	at com.alipay.oceanbase.rpc.direct_load.ObDirectLoadConnection$Builder.build(ObDirectLoadConnection.java:318)
	at com.oceanbase.spark.directload.DirectLoaderBuilder.buildConnection(DirectLoaderBuilder.java:191)
	at com.oceanbase.spark.directload.DirectLoaderBuilder.build(DirectLoaderBuilder.java:163)
	... 14 more

```

### Environment

1.2.14-SNAPSHOT

### Fast reproduce steps

Try calling direct load API multiple times. The above error will appear.

### Expected behavior

_No response_

### Actual behavior

_No response_

### Additional context

_No response_

--- a/src/main/java/com/alipay/oceanbase/rpc/bolt/transport/ObTableConnection.java
+++ b/src/main/java/com/alipay/oceanbase/rpc/bolt/transport/ObTableConnection.java
@@ -173,8 +173,12 @@ public class ObTableConnection {
                     credential = result.getCredential();
                     tenantId = result.getTenantId();
                     // Set version if missing
-                    if (ObGlobal.obVsnMajor() == 0 && !result.getServerVersion().isEmpty()) {
+                    if (ObGlobal.obVsnMajor() == 0) {
                         // version should be set before login when direct mode
+                        if (result.getServerVersion().isEmpty()) {
+                            throw new RuntimeException(
+                                "Failed to get server version from login result");
+                        }
                         LocationUtil.parseObVerionFromLogin(result.getServerVersion());
                         LOGGER.info("The OB_VERSION parsed from login result is: {}",
                             ObGlobal.OB_VERSION);

